### DISCUSSION TITLE
Missing repositories in Electric dev capital report data

### DISCUSSION THREAD

**Discussion by AbdelStark** [stark-rocket/.github#9] [Ideas]
This [repo](https://github.com/electric-capital/crypto-ecosystems/blob/master/data/ecosystems/s/starknet.toml) references open source repositories that are related to Starknet ecosystem.
Let's identify those who are missing and submit a pull request to fill the gap.

If one or more of your project repositories are missing in the [Starknet file](https://github.com/electric-capital/crypto-ecosystems/blob/master/data/ecosystems/s/starknet.toml) please comment and add links to all repos. And really ALL open source repos related to Starknet, Cairo smart contracts, sdks, libraries, frontend, backend, infra, documentation,...

We will then submit a pull request to fill the gap.

**Comment by onurinanc** [comment]
- A STARK-friendly elliptic curve designed to work within Cairo circuits on Starknet: https://github.com/hashcloak/starkjub
- Stark Friendly Privacy Primitives in Cairo: https://github.com/hashcloak/Starkacy/
- AES Encryption and Decryption implemented in Cairo: https://github.com/onurinanc/cairo-aes

**Comment by cannsky** [comment]
StarkLearn educational content codes: https://github.com/project3fusion/StarkLearn
StarkSharp (C# library for Starknet): https://github.com/project3fusion/StarkSharp
StarkMoba - open source moba game based on Starknet: https://github.com/project3fusion/StarkMOBA
FusionFPS - open source fps game based on Starknet: https://github.com/project3fusion/FusionFPS

**Comment by raphaelDkhn** [comment]
* Orion ZKML library: https://github.com/gizatechxyz/orion
* Orion ZKML tutorials: https://github.com/gizatechxyz/orion_tutorials

**Comment by faytey** [comment]
An api endpoint for getting gas cost of declared and deployed contracts on starknet: [https://github.com/faytey/starknet-gasprice-gas-limit-endpoint/tree/master](https://github.com/faytey/starknet-gasprice-gas-limit-endpoint/tree/master)

**Comment by rkdud007** [comment]
- cairo lib - comprehensive library for the Cairo language: https://github.com/HerodotusDev/cairo-lib
- cairo mmr - cairo merkle mountain range: https://github.com/HerodotusDev/cairo-mmr
- starksight - verify world id on starknet: https://github.com/HerodotusDev/starksight-zkhack

- dojo chess - chess game made with dojo: https://github.com/rkdud007/chess-dojo

**Reply by rkdud007** [reply]
Adding one more just public
-  offchain evm headers processor : This repository implements the logic of building and maintaining two Merkle Mountain Ranges (MMRs) containing only provably valid Ethereum block headers using cairo programs- https://github.com/HerodotusDev/offchain-evm-headers-processor

**Comment by PhilippeR26** [comment]
- a tutorial to use Starknet.js with Typescript : https://github.com/PhilippeR26/starknet.js-workshop-typescript
- a tool to store/read small binaries in Starknet : https://github.com/PhilippeR26/file2felts

**Comment by 0xKawaka** [comment]
BlockHeroes, fully on-chain turn based rpg : https://github.com/0xKawaka/BlockHeroesCairo

**Comment by YohanTz** [comment]
Zohal's front end, perpetual Exchange on Starknet: https://github.com/Zohal-Starknet/zohal-interface

**Comment by Kelvyne** [comment]
Starknet storage proof in Solidity : https://github.com/Kelvyne/starknet-storage-proof-solidity

**Comment by Akinbola247** [comment]
Fully onchain tic-tac-toe game : https://github.com/Akinbola247/Tictactoe-Starknet

**Comment by ivpavici** [comment]
These 3 repos have moved:

from: https://github.com/Shard-Labs/starknet-hardhat-example -> to: https://github.com/0xSpaceShard/starknet-hardhat-example

from: https://github.com/Shard-Labs/starknet-hardhat-plugin -> to: https://github.com/0xSpaceShard/starknet-hardhat-plugin

from: https://github.com/Shard-Labs/cairo-cli-docker -> to: https://github.com/0xSpaceShard/cairo-cli-docker

**Comment by lferrigno** [comment]
Cairo Native: https://github.com/lambdaclass/cairo_native

**Comment by devnet0x** [comment]
GM..please add this links:

Writeup about L1-L2 messaging (Cairo 0): https://github.com/devnet0x/Blockchain/tree/master/Cairo/starknet_messaging_bridge
Spanish article about Cairo Security: https://medium.com/@devnet0x/una-introducci%C3%B3n-a-la-seguridad-de-los-contratos-inteligentes-en-cairo-1-0-6bc7d2ab23b0
Bash script for protostar: https://github.com/devnet0x/Proto_build
Basecamp Cohort2 homework solutions (Cairo 0): https://github.com/devnet0x/Starknet_basecamp_cohort2

I dont know if this helps. Forks of projects where i have been contributing:

Starknet Foundry: https://github.com/devnet0x/starknet-foundry
Starknet Book: https://github.com/devnet0x/starknetbook
Starknet Homepage: https://github.com/devnet0x/starknet_homepage
Kakarot: https://github.com/devnet0x/kakarot

**Comment by gaetbout** [comment]
GM,
 - Cairo0 Array manipulation lib: https://github.com/gaetbout/starknet-array-manipulation
 - Cairo0 Felt packing lib: https://github.com/gaetbout/starknet-felt-packing
 - Commit reveal example: https://github.com/gaetbout/starknet-commit-reveal (will be updating it to latest soon)
 - ERC20 Lazy init: https://github.com/gaetbout/starknet-erc20-lazy-initialization
 - Stark utils: https://github.com/gaetbout/stark-utils
 - StarkWhale alert (Twitter bot): https://github.com/gaetbout/StarkWhaleAlert

**Comment by shramee** [comment]
WASM blockifier (WIP): https://github.com/shramee/wasm-blockifier-runtime
Hanji (docs generator for Cairo): https://github.com/shramee/hanji
Cairo functions catalog (Hanji + OpenAI + MongoDB Lucene): https://github.com/shramee/cairolatte-backend
Dojo game utils (incomplete game): https://github.com/shramee/dojo-gems
(Predictable) Randomness in Cairo: https://github.com/shramee/cairo-random
P5 JS starter for Dojo: https://github.com/shramee/dojo-p5-starter
Raw JS starter with Dojo utils: https://github.com/shramee/dojo-raw-starter
Linear ERC20 equivalent (tokens can't be Copied but only moved): https://github.com/shramee/linear-erc20
Early ECS implementation in Cairo 0 (December 22): https://github.com/shramee/cairo-ecs

**Comment by glihm** [comment]
Ark project: https://github.com/ArkProjectNFTs/ark-project
Katana-ci: https://github.com/ArkProjectNFTs/katana-ci
Messaging testing: https://github.com/glihm/starknet-messaging-dev

**Comment by cairofiofficial** [comment]
CairoFi is a V3 AMM on Starknet:  https://github.com/cairofiofficial

**Comment by StarkIDreal** [comment]
StarkID is a one-stop web3.0 domain service provider on Starknet https://github.com/StarkIDreal

**Comment by Stark-Wars** [comment]
full on-chian game on starknet: https://github.com/Stark-Wars/StarkWarsContracts

**Comment by cantlee** [comment]
Starkdex AMM:Liquidity Farms,and Reward pools on Starknet-https://github.com/cantlee

**Comment by FrostStarBook** [comment]
## crypts and caverns(Map component, in the ETH ecosystem and StarkNet ecosystem.)
- 9000 generative on-chain dungeons in the Lootverse: https://github.com/CheDAOLabs/cryptsandcaverns

- This is a demonstration project for C&C (crypts and caverns). Its purpose is to fetch detailed map information of NFTs from the blockchain within the C&C contract. Then, it utilizes the Unity engine to render the map into a game scene that can be explored and played in a realistic manner. https://github.com/CheDAOLabs/cryptsandcaverns-game-demo

- An Example projects of integrating C&C maps developed by the CHE-DAO team with Dojo.https://github.com/CheDAOLabs/cc-dojo-map

## full on-chain games in the StarkNet ecosystem.
- A full on-chain adventure magic world game.  https://github.com/CheDAOLabs/CheWorld

- CheCell is an evolutionary onchain game featuring living cells that players can customize,evolve,adn breed. Built based on Dojo.https://github.com/CheDAOLabs/CheCell

- A full on-chain SLG (simulative life games) game set in the steam (industrial) era. https://github.com/CheDAOLabs/che-verse

## Library(cairo)
- A collection of tools in Cairo language development.https://github.com/CheDAOLabs/torch

**Comment by Orland0x** [comment]
Repos from the Snapshot Labs team relating to Snapshot X and the Checkpoint Indexer: 
https://github.com/snapshot-labs/sx.js
https://github.com/snapshot-labs/sx-starknet
https://github.com/checkpoint-labs/checkpoint
https://github.com/checkpoint-labs/checkpoint-template
https://github.com/snapshot-labs/sx-api

**Comment by rohitgoyal** [comment]
GM
Here are some of the JediSwap repos
https://github.com/jediswaplabs/JediSwapE2-interface
https://github.com/jediswaplabs/salary-contract
https://github.com/jediswaplabs/contributors-SBT
https://github.com/jediswaplabs/missions-interface
https://github.com/jediswaplabs/JediSwap
https://github.com/jediswaplabs/contest-nft
https://github.com/jediswaplabs/analytics-dashboard

**Comment by clexmond** [comment]
Missing cubit (fixed point math library): https://github.com/influenceth/cubit

**Comment by Astro-swap** [comment]
AstroSwap:AMM,Liquidity Farms,and Reward pools on Starknet-https://github.com/Astro-swap

**Comment by kfastov** [comment]
My small contribution: general purpose ERC721 contract with metadata:
https://github.com/kfastov/starknet-erc721-metadata
